src/tutorials/abilities_advanced.md

```diff
@@ -174,17 +174,8 @@ Use the **[VFX section](weapons.md#adding-visual-effects)** of the simple weapon
         Type the below code to create variables for the `equipment` and the `ability`:
 
         ```lua
-        local EQUIPMENT = script:FindAncestorByType('Equipment')
-
-        if not EQUIPMENT:IsA('Equipment') then
-            error(script.name .. " should be part of Equipment object hierarchy.")
-        end
-
-        local ABILITY = script:FindAncestorByType('Ability')
-
-        if not ABILITY:IsA('Ability') then
-            error(script.name .. " should be part of Ability object hierarchy.")
-        end
+        local EQUIPMENT = assert(script:FindAncestorByType('Equipment'), script.name .. " should be part of Equipment object hierarchy.")
+        local ABILITY = assert(script:FindAncestorByType('Ability'), script.name .. " should be part of Ability object hierarchy.")
```

Comment:
It's a tutorial. The whole :IsA() should probably be removed. Adding assert() makes it more complex than what we are trying to teach. Remove both :IsA() blocks and the resulting code would be just:
local EQUIPMENT = script:FindAncestorByType('Equipment')
local ABILITY = script:FindAncestorByType('Ability')

Reply:
Hard agree on nixing :IsA() where possible. Validations are good practice, but from an instructional perspective, if they are necessary, they need to be introduced in a separate step, after the initial relevant code.